### PR TITLE
Add scraping functionality for Meta

### PR DESCRIPTION
**Pull request by lilyxmeng**
#### What is the purpose of your change?
Add scraping code for Meta.

#### What areas of the application does this change impact? (changes API, folder structure, etc)
scraper/main.py

#### What features/fixes are deferred/not handled in this PR?
The code worked previously until it randomly started opening the page and saying I was not logged in -- do we want to create a random account to login with to ensure we don't run into this issue during deployment? In that case, I would have to look further into how to enter the login for a page via selenium.

## Checklist

### CODE REVIEW

**Comment by jwu910** [comment]
@lilyxmeng Can you rebase on to bobatalks(upstream)/main ? All of the commits from your other branch are present here

I recommend also spending time to get more familiar with Git and version control, it will be incredibly helpful get used to writing meaningful commit messages and building that habit now, makes reading and understanding changes and the intention behind changes easier. 

Easy way to remember is to write in the imperative mood. In your head you can say "if merged, this commit will:" and the following statement can be your commit message.

**Review by jwu910 — CHANGES_REQUESTED** [review]
Was this ready for review btw? I saw you mentioned it stopped working.

I suggest you gather details about this and communicate with the group chat to try and get some help and collaborate to find a solution, maybe someone else may have an idea as to why this is unable to scrape correctly

File: scraper/main.py [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+    "Apple": "https://jobs.apple.com/en-ca/search?location=canada-CANC+united-states-USA+mexico-MEXC&team=internships-STDNT-INTRN",
+    "Netflix": "https://jobs.netflix.com/search?q=internship",
+    "Google": "https://careers.google.com/jobs/results/?q=internship",
+}
+
+
+def get_meta_internships(soup):
+    driver.get(urls["Meta"])
+    
+    containers = driver.find_elements("xpath", "//div[@class='_6g3g x8y0a91 x7z1be2 xngnso2 xeqmlgx xo1l8bm xeqr9p9 x1e096f4']")
```
[the remark this answers is not in the document]
Comment (lilyxmeng) [reply scraper/main.py:20]: I'm not sure, all the classes on the page look randomly generated.

[the remark this answers is not in the document]
Comment (a-camarillo) [reply scraper/main.py]: @jwu910 I just took a look at this and the class names look pretty much the same(randomly generated?). The divs containing basic career information we do need do have the same class name "x1ypdohk" but I'm not sure how consistent we can consider this   
image

[the remark this answers is not in the document]
Comment (jwu910) [reply scraper/main.py]: I would want to bet that these are just namespaced and generated. I would not rely on them as consistent selectors. Can we check a level or two up to find something consistent and generic? 

The goal is to identify patterns that lead us to the data we want right?

Like is `div[role=link]` as a direct descendant of whatever is above that going to be unique enough to get us the data we need?

[the remark this answers is not in the document]
Comment (a-camarillo) [reply scraper/main.py]: I don't think the levels above are any better than what we currently have. There is a div with `id=careersContentContainer` but that is for the entire page element containing the career divs. Anything in between is seemingly generated.
image

[the remark this answers is not in the document]
Comment (jwu910) [reply scraper/main.py]: Is this structure the same for every job posting?

Cause if the constant is the ID, we technically could build out the selector with elements up to the div with the matching role right?

[the remark this answers is not in the document]
Comment (a-camarillo) [reply scraper/main.py]: ID is the constant for the page but each posting is shown by the `class="x1ypdohk"`, hopefully this screenshot isn't too small but for each job posting are more inner divs of generated classes to get the info we want. I suppose we can still use these class names to get the info we want but we will still lack the certainty we want

image

[the remark this answers is not in the document]
Comment (jwu910) [reply scraper/main.py]: could this be something like 

`div#careersContentContainer > div > div > div:nth-child(2) > div:nth-child(3)` or something similar?

[the remark this answers is not in the document]
Comment (jwu910) [reply scraper/main.py]: @lilyxmeng thoughts?

[the remark this answers is not in the document]
Comment (lilyxmeng) [reply scraper/main.py]: I'll test something like this out and let you know of any changes.

[the remark this answers is not in the document]
Comment (jwu910) [reply scraper/main.py]: Sounds good!

Was skimming over some articles to research a little and there were some mention of running the web drivers in headless mode as well, is that something we're already doing? Idk if its going to solve any of the current issues, but supposedly helpful when working with scraping SPAs at least

**Comment by jwu910** [comment]
Will be addressed in another PR.

CC: @a-camarillo

**Pull request closed without merging**